products/urls.py

```diff
@@ -0,0 +1,18 @@
+from django.urls import path
+
+from . import views
+
+urlpatterns = [
+    path("homepage/", views.homepage, name="homepage"),
+    path('<str:category_name>/', views.category_product_page, name='category_products')
+]
+
+""" path("mobiles/", views.category_product_page, name="mobiles"),
```

Comment:
you didn't need to repeat this for each category.

https://forum.djangoproject.com/t/passing-form-get-data-in-url-tag/13060/2

Read more in docs